third_party/winuwp_h264/H264Encoder/H264Encoder.cc

```diff
@@ -55,6 +60,63 @@ WinUWPH264EncoderImpl::~WinUWPH264EncoderImpl() {
   Release();
 }
 
+namespace {
+HRESULT MakeMediaTypeOut(UINT32 width,
+                         UINT32 height,
+                         UINT32 target_bps,
+                         UINT32 frame_rate,
+                         IMFMediaType** res_out) {
+  HRESULT hr = S_OK;
+  // output media type (h264)
+  ON_SUCCEEDED(MFCreateMediaType(res_out));
+  auto media_type_out = *res_out;
+  ON_SUCCEEDED(media_type_out->SetGUID(MF_MT_MAJOR_TYPE, MFMediaType_Video));
+  ON_SUCCEEDED(media_type_out->SetGUID(MF_MT_SUBTYPE, MFVideoFormat_H264));
+  // Lumia 635 and Lumia 1520 Windows phones don't work well
+  // with constrained baseline profile.
+  // todo(fibann): this is obsolete and should probably use the profile passed
+  // in the settings
+  // ON_SUCCEEDED(media_type_out->SetUINT32(MF_MT_MPEG2_PROFILE,
+  // eAVEncH264VProfile_ConstrainedBase));
+
+  ON_SUCCEEDED(media_type_out->SetUINT32(MF_MT_AVG_BITRATE, target_bps));
+  ON_SUCCEEDED(media_type_out->SetUINT32(MF_MT_INTERLACE_MODE,
+                                         MFVideoInterlace_Progressive));
+  ON_SUCCEEDED(MFSetAttributeSize(media_type_out, MF_MT_FRAME_SIZE, width,
+                                  height));
+  ON_SUCCEEDED(MFSetAttributeRatio(media_type_out, MF_MT_FRAME_RATE,
+                                   frame_rate, 1));
+  return hr;
+}
+
+HRESULT SetMediaTypeIn(const ComPtr<IMFSinkWriter>& sink_writer,
+                    DWORD stream_index,
+                    UINT32 width,
+                    UINT32 height,
+                    UINT32 frame_rate) {
+  HRESULT hr = S_OK;
+  // input media type (nv12)
+  ComPtr<IMFMediaType> media_type_in;
+  ON_SUCCEEDED(MFCreateMediaType(&media_type_in));
+  ON_SUCCEEDED(media_type_in->SetGUID(MF_MT_MAJOR_TYPE, MFMediaType_Video));
+  ON_SUCCEEDED(media_type_in->SetGUID(MF_MT_SUBTYPE, MFVideoFormat_NV12));
+  ON_SUCCEEDED(media_type_in->SetUINT32(MF_MT_INTERLACE_MODE,
+                                        MFVideoInterlace_Progressive));
+  ON_SUCCEEDED(media_type_in->SetUINT32(MF_MT_ALL_SAMPLES_INDEPENDENT, TRUE));
```

Comment:
>TRUE [](start = 71, length = 4)

Shouldn't this be FALSE?

[https://docs.microsoft.com/en-us/windows/win32/medfound/mf-mt-all-samples-independent-attribute](https://docs.microsoft.com/en-us/windows/win32/medfound/mf-mt-all-samples-independent-attribute)

If this attribute is FALSE, some samples cannot be used without referring to other samples in the stream. 
*** For example, if a video format contains delta frames, this attribute should be FALSE. ***

Reply:
Never mind NV12 is a raw format.

---
In reply to: [344866673](https://github.com/webrtc-uwp/webrtc-windows/pull/70#discussion_r344866673) [](ancestors = 344866673)